src/main/scala/com/databricks/spark/avro/DefaultSource.scala

```diff
@@ -16,16 +16,210 @@
 
 package com.databricks.spark.avro
 
-import org.apache.spark.sql.SQLContext
-import org.apache.spark.sql.sources.{HadoopFsRelation, HadoopFsRelationProvider}
+import java.io.{FileNotFoundException, IOException, ObjectInputStream, ObjectOutputStream}
+import java.net.URI
+import java.util.zip.Deflater
+
+import scala.util.control.NonFatal
+
+import com.databricks.spark.avro.DefaultSource.{IgnoreFilesWithoutExtensionProperty, SerializableConfiguration}
+import org.apache.avro.SchemaBuilder
+import org.apache.avro.file.{DataFileConstants, DataFileReader}
+import org.apache.avro.generic.{GenericDatumReader, GenericRecord}
+import org.apache.avro.mapred.{AvroOutputFormat, FsInput}
+import org.apache.avro.mapreduce.AvroJob
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileStatus, Path}
+import org.apache.hadoop.mapreduce.Job
+import org.slf4j.LoggerFactory
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.catalyst.expressions.GenericRow
+import org.apache.spark.sql.execution.datasources.{FileFormat, OutputWriterFactory, PartitionedFile}
+import org.apache.spark.sql.sources.{DataSourceRegister, Filter}
 import org.apache.spark.sql.types.StructType
 
-private[avro] class DefaultSource extends HadoopFsRelationProvider {
+private[avro] class DefaultSource extends FileFormat with DataSourceRegister {
```

Comment:
Tried renaming, and hit test failures.

The problem is that, if users are using fully qualified *package* name (i.e. "com.databricks.spark.avro" in the case of spark-avro), then the `DataSourceRegister` class must be named as `DefaultSource`. For backwards compatibility consideration. I'd prefer not changing it.